src/main/scala/edu/berkeley/ce/rockslicing/BlockVTK.scala

```diff
@@ -0,0 +1,188 @@
+package edu.berkeley.ce.rockslicing
+
+import breeze.linalg
+import breeze.linalg.{DenseVector, DenseMatrix}
+
+object BlockVTK {
+
+  /**
+    * Finds the center of a list of vertices - defined as the average coordinate of all the vertices in the list
+    * @param vertices List of vertices
+    * @return Coordinate of center of vertices
+    */
+  private def findCenter(vertices: Seq[(Double, Double, Double)]): (Double, Double, Double) = {
+    (vertices.map(_._1).sum / vertices.length.toDouble,
+     vertices.map(_._2).sum / vertices.length.toDouble,
+     vertices.map(_._3).sum / vertices.length.toDouble)
+  }
+
+  /**
+    * Compares pointA to pointB. If pointA is first relative to pointB rotating counter-clockwise about center,
+    * the method returns true. Input vectors should have same z-coordinate - comparison is based in 2-D
+    * @param pointA Point of interest
+    * @param pointB Point of comparison
+    * @param center Reference point for comparison
+    * @return Returns TRUE if pointA is first relative to pointB, FALSE otherwise
+    */
+  private def ccwCompare(pointA: (Double, Double, Double), pointB: (Double, Double, Double),
+                         center: (Double, Double, Double)): Boolean = {
+    // Check that points are in the same x-y plane
+    try {
+      if (math.abs(pointA._3 - pointB._3) > NumericUtils.EPSILON) {
+        throw new IllegalArgumentException("Exception thrown")
```

Comment:
What I'm saying is that you should write something like this:
```scala
private def ccwCompare(...): Boolean = {
  if (math.abs(pointA._3 - pointB._3) > NumericUtils.EPSILON) {
    throw new IllegalArgumentException("Input points are not in the same plane")
  }

if ((pointA._1 - center._1  ...
// Just keep the rest of the code the same
```

Throwing and catching your own exception doesn't really do a whole lot. The point of an exception is that you communicate to whoever called this method that something went wrong. The caller can catch an exception if they choose (or if they have to in Java, but that's another issue) and then respond appropriately, or if they do not catch the exception then execution the user will get a helpful stack trace printed out.